alica_engine/include/engine/util/HashFunctions.h

```diff
@@ -2,16 +2,42 @@
 #include <stdint.h>
 
 #include "engine/util/cityhash.h"
+#include <functional>
 
 // Thin wrapper over the used hash function:
 namespace alica
 {
 
-// TODO: implement a hash function for integers that avoids collissions
-std::size_t contextHash(int64_t value);
+template <class T>
+struct dependent_false : std::false_type
+{
+};
 
-// TODO: implement a function that can combine 2 hashes
-std::size_t contextHashCombine(std::size_t h1, std::size_t h2);
+template <class T>
+inline void hash_combine(std::size_t& seed, const T& value)
+{
+    // Simple hash_combine, see e.g. here:
+    // https://github.com/HowardHinnant/hash_append/issues/7
+    // Not sure we ever need 32bit, but here it is...
+    if constexpr (sizeof(std::size_t) == 4) {
+        seed ^= std::hash<T>{}(value) + 0x9e3779b9U + (seed << 6) + (seed >> 2);
+    } else if constexpr (sizeof(std::size_t) == 8) {
+        seed ^= std::hash<T>{}(value) + 0x9e3779b97f4a7c15LLU + (seed << 12) + (seed >> 4);
+    } else {
+        static_assert(dependent_false<T>::value, "hash_combine not implemented");
+    }
+}
+
+std::size_t contextHash(int64_t value)
+{
+    return std::hash<int64_t>{}(value);
```

Comment:
can you add a comment here that libstdc++ uses murmur hash & so its okay to use std::hash. Otherwise this function needs to be re-implemented on other architectures to use a suitable hash function